mcux/mcux-sdk/devices/MIMX9352/drivers/fsl_clock.h

```diff
@@ -1176,6 +1176,24 @@ static inline void CLOCK_SetRootClockMux(clock_root_t root, uint8_t src)
     __ISB();
 }
 
+/*!
+ * @brief Set CCM Root Clock MUX node to certain value.
+ *
+ * @param regBase Physical/Virtual base address of CCM.
+ * @param root Which root clock node to set, see \ref clock_root_t.
+ * @param src Clock mux value to set, different mux has different value range. See \ref clock_root_mux_source_t.
+ */
+static inline void CLOCK_SetRootClockMuxMapped(uint32_t *regBase, clock_root_t root, uint8_t src)
+{
+    CCM_Type *ccm_base = (CCM_Type *)regBase;
+
+    assert(src < 8U);
+    ccm_base->CLOCK_ROOT[root].CLOCK_ROOT_CONTROL.RW =
+        (ccm_base->CLOCK_ROOT[root].CLOCK_ROOT_CONTROL.RW & ~(CCM_CLOCK_ROOT_MUX_MASK)) | CCM_CLOCK_ROOT_MUX(src);
+    __DSB();
+    __ISB();
+}
+
```

Comment:
This should be a PR to the SDK code. We do not want to maintain changes to SDK code in the Zephyr HAL repo.